### ISSUE TITLE
Posts with more characters than twitter allows can't be crossposted

### ISSUE THREAD

**Issue by Waithamai** [diaspora/diaspora#7714]
In the past, it was possible to crosspost status messages with more than 140 - or 280 - characters to twitter as well. they were cut after the allowed number of characters and a link to the diaspora post was added. Now they never appear on twitter. (Crossposting still works for posts with fewer characters.)

**Comment by Waithamai** [comment]
Uhm. So I just noticed crossposting actually worked for a long post. I then tested with some other posts and it seems crossposting **always** works if the beginning of the post which will be displayed in the tweet contains an url. If there's no url, crossposting always fails. Plain url or with markdown doesn't make any difference.

**Comment by HankG** [comment]
I confirm this behavior and see what's happening in the code.  I can attempt to fix this in a branch for confirmation of behavior and add more rspec tests to reflect this.

**Comment by SuperTux88** [comment]
> Should I look at coming up with some sort of tests to directly test that behavior

Yes, a normal test should do it.

> I haven't had any pull requests ever successfully integrated on this project and already have one in the pipeline so rather than do another pull request I'm just going to post the link to the branch in my repo

It's not problem to have multiple pull requests open and it's much easier to review and give feedback on pull requests, so please open a PR for this (but can you please squash the commits before opening the PR).

**Comment by silberzwiebel** [comment]
Unsure whether this fits here but I observed the following when crossposting to twitter: The very same text (incl. one URL) for which Twitter says, I still have 4 characters left is displayed as already 33 characters too much in diaspora*'s UI. Crossposting still did work fine. I think it's because diaspora* counts all characters but does not recognize that URLs in the post are treated with a fixed length at twitter, no matter how long the actual URL is. Is this issue known and potentially already solved with the upcoming pull request? Thanks!

**Comment by HankG** [comment]
@silberzwiebel yes it is related but it is different.  When I was working on the backend code I noticed that the UI is not doing the shortening logic to approximate the behavior in Twitter.  I have a pull request I just submitted for the backend fix but I didn't attempt to address the UI one.  I was thinking of doing maybe trying to tackle that in another change branch but didn't have time to tackle it now.
